cfpb/cfgov-refresh
# Add option to filter Related Posts by ALL tags on host page, rather than ANY ONE of the tags

The need driving this work is to be able to show only posts about a specific topic that are written in Spanish. Currently, the only way to know if a blog post is in Spanish is that it has a "CFPB en español" tag. Therefore, we need to match both that tag and the topic tag.


## Additions

- Option on Related Posts module to tell it to use `AND`-based filtering
- Logic to `CFGOVPage.related_posts` that looks for said option and filters accordingly


## Testing

_Ensure that you have a DB dump newer than Sept. 13, 2016, or you will have to create a test blog post with topics "CFPB en español" and "Banking"._

1. Pull branch, migrate, runserver
1. Add a related posts module to a page, making sure to check the "Match all topics this page is tagged with" option
1. If not using a database dump newer than Sept. 13, 2016, Create a test blog post with the same topics, and test posts with only one of each of those two topics.
1. Set two topics on the page in its Settings tab: "CFPB en español" and "Banking"
1. View the page and see that the Related Posts module is returning only one post, the one with both of the same topics as the page.

## Screenshots

![screen shot 2017-04-20 at 16 46 52](https://cloud.githubusercontent.com/assets/1044670/25252003/6bd898d0-25e9-11e7-8267-9d4a618d9a03.png)


## Notes

- I did not add any tests for this new functionality, due to refactoring discussions happening in GHE/platform#825 


## Checklist
